rama-http-types/src/curl/traits.rs

```diff
@@ -0,0 +1,55 @@
+use crate::{Body, Request, Parts}
+use crate::dep::http_body;
+use bytes::Bytes;
+use rama_core::error::{BoxError, OpaqueError};
+
+mod sealed {
+    pub trait Sealed {}
+
+    impl<B> Sealed for crate::Request<B> 
+    where 
+    B: http_body::Body<Data = Bytes, Error = Into<BoxError>> + Send + Sync 
+    {}
+
+    impl Sealed for crate::Parts {}
+
+    impl<T: Sealed> Sealed for &T {}
+}
+
+/// Sealed trait for converting HTTP request headers into a curl command
+
+pub trait IntoCurlHeadersCommand: sealed::Sealed {
+    fn into_curl_headers_command(&self) -> Result<String, OpaqueError>;
+}
+
+
+/// Sealed trait for converting a full HTTP request into a curl command.
+pub trait IntoCurlCommand: sealed::Sealed {
+    fn into_curl_command(self) -> Result<(String, Request), OpaqueError>;
```

Comment:
it would be a bit silly I think (correct me if mistaken) to have to format the cmd for each impl. As such I think it's more useful if this just provides access to the relevant parts.

E.g. `IntoCurlCommandPriv` can be done with just a single method: `async fn try_into_request(self) -> Result<Request, OpaqueError>`